spec/src/modules/catalog.js

```diff
@@ -0,0 +1,2625 @@
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/items.csv');
+        const itemsStream = fs.createReadStream(filePath);
+        const data = {
+          items: itemsStream,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of items using buffers', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/items.csv');
+        const itemsBuffer = fs.readFileSync(filePath);
+        const data = {
+          items: itemsBuffer,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of variations using streams', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/variations.csv');
+        const variationsStream = fs.createReadStream(filePath);
+        const data = {
+          variations: variationsStream,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of variations using buffers', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/variations.csv');
+        const variationsBuffer = fs.readFileSync(filePath);
+        const data = {
+          variations: variationsBuffer,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of item groups using streams', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/item_groups.csv');
+        const itemGroupsStream = fs.createReadStream(filePath);
+        const data = {
+          item_groups: itemGroupsStream,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of item groups using buffers', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        const filePath = path.join(process.cwd(), './spec/src/csv/item_groups.csv');
+        const itemGroupBuffer = fs.readFileSync(filePath);
+        const data = {
+          item_groups: itemGroupBuffer,
+          section: 'Products',
+        };
+
+        catalog.replaceCatalog(data).then(done);
+      });
+
+      it('Should replace a catalog of item groups using streams', (done) => {
```

Comment:
Looks like this is a duplicate test, it exists on line 2366.

Reply:
I was trying to make it look like I wrote more tests. Caught